nf_core/modules/modules_json.py

```diff
@@ -345,37 +345,42 @@ def unsynced_modules(self):
             if module_repo_name is not None:
                 # If it does, check if the module is in the 'modules.json' file
                 module = str(dir.relative_to(module_repo_name))
+                module_repo = missing_installation[module_repo_name]
 
-                if module not in missing_installation[module_repo_name].get("modules", {}):
+                if module not in module_repo.get("modules", {}):
                     untracked_dirs.append(dir)
                 else:
                     # Check if the entry has a git sha and branch before removing
-                    modules = missing_installation[module_repo_name]["modules"]
+                    modules = module_repo["modules"]
                     if "git_sha" not in modules[module] or "branch" not in modules[module]:
-                        raise UserWarning(
-                            "The 'modules.json' file is not up to date. "
-                            "Please reinstall it by removing it and rerunning the command."
+                        self.determine_module_branches_and_shas(
+                            module, module_repo["git_url"], module_repo["base_path"], [module]
                         )
-                    missing_installation[module_repo_name]["modules"].pop(module)
-                    if len(missing_installation[module_repo_name]["modules"]) == 0:
+                    module_repo["modules"].pop(module)
+                    if len(module_repo["modules"]) == 0:
                         missing_installation.pop(module_repo_name)
             else:
                 # If it is not, add it to the list of missing modules
                 untracked_dirs.append(dir)
 
         return untracked_dirs, missing_installation
 
-    def has_git_url_and_base_path(self):
+    def has_correct_format(self):
         """
         Check that that all repo entries in the modules.json
-        has a git url and a base_path
+        have a git url and a base_path
 
         Returns:
             (bool): True if they are found for all repos, False otherwise
         """
         for repo_entry in self.modules_json.get("repos", {}).values():
             if "git_url" not in repo_entry or "base_path" not in repo_entry:
-                return False
```

Comment:
This function now doesn't have a consistent return value any more. Also, the function name became less descriptive of the procedure inside the function. Both issues make the code harder to read and should ideally be avoided.

Reply:
I reverted the changes now again a bit and should be clearer (as soon as we are done with https://github.com/nf-core/tools/issues/1732